Documentation/proposals/202305-scrapeclasses.md

```diff
@@ -0,0 +1,189 @@
+
+## Scrape Classes
+
+* **Owners:**
+  * `eronwright`
+
+* **Related Tickets:**
+  * https://github.com/prometheus-operator/prometheus-operator/issues/4121
+  * https://github.com/prometheus-operator/prometheus-operator/issues/3922
+
+This proposal aims to introduce the concept of _scrape classes_, to enable users to leverage scrape configuration data 
+that is provided by the administrator. For example, to allow pod monitors and probes to safely use
+unsafe TLS configuration fields (e.g. `keyFile`). 
+
+## Why
+
+Today, there are some scenarios that are difficult to support in a safe way, notably the need to reference 
+a local TLS file in a scrape configuration. For example, to scrape a pod in an Istio mesh with strict mTLS would require that
+the scrape configuration use a TLS keyfile that is provided by the Prometheus server's Istio sidecar. 
+See [Istio documentation](https://istio.io/latest/docs/ops/integrations/prometheus/#tls-settings) for more details.
+
+Another motivation is to improve feature parity amongst the monitor resources. The `PodMonitor` and `Probe` resources
+aren't at parity with `ServiceMonitor` because the latter allows for unsafe TLS settings.
+
+A minor motivation is to decouple the monitor spec from low-level infrastructure details like TLS certificates.
+
+### Pitfalls of the current solution
+
+The only known solution for use cases where you'd need to use unsafe TLS settings is to use `additionalScrapeConfig`. 
```

Comment:
I revised the section to mention `ScrapeConfig`.

Reply:
Would you suggest that `ScrapeConfig` be mentioned in this document? Could you provide a snippet to be included?

Reply:
It could be generic, keeping the `tlsConfig` example in mind:
```yaml
apiVersion: monitoring.coreos.com/v1alpha1
kind: ScrapeConfig
metadata:
  name: scrape-config
spec:
  scrapeClass: istio-tls-etc
  staticConfigs:
    [...]
  httpSDConfig:
    [...]
  fileSDConfig:
    [...]
```

Taking my second comment into account, it could also define the different files from `fileSDConfig` as well as endpoints from `httpSDConfig`. At the moment though I'm only opening the discussion and want to see others' feedback as to extending the `ScrapeClass` to do more than just `tlsConfig` would make sense.